src/rules/no-export-load-in-svelte-module-in-kit-pages.ts

```diff
@@ -0,0 +1,51 @@
+import type * as ESTree from "estree"
+import { createRule } from "../utils"
+import { isKitPageComponent } from "../utils/svelte-kit"
+
+export default createRule("no-export-load-in-svelte-module-in-kit-pages", {
+  meta: {
+    docs: {
+      description:
+        "disallow exporting load functions in `*.svelte` module in Svelte Kit page components.",
+      category: "Possible Errors",
+      // TODO Switch to recommended in the major version.
+      recommended: false,
+    },
+    schema: [],
+    messages: {
+      unexpected:
+        "disallow exporting load functions in `*.svelte` module in Svelte Kit page components.",
+    },
+    type: "problem",
+  },
+  create(context) {
+    if (!isKitPageComponent(context)) {
+      return {}
+    }
+    let isModule = false
+    return {
+      // <script context="module">
+      [`Program > SvelteScriptElement > SvelteStartTag > SvelteAttribute[key.name="context"] > SvelteLiteral[value="module"]`]:
+        () => {
+          isModule = true
+        },
+
+      // </script>
+      "Program > SvelteScriptElement:exit": () => {
+        isModule = false
+      },
+
+      // export function load() {}
+      // export const load = () => {}
+      [`ExportNamedDeclaration :matches(FunctionDeclaration, VariableDeclaration > VariableDeclarator) > Identifier[name="load"]`]:
```

Comment:
Could you change the selector to match more precisely? (I thought I sent it, but it looks like I missed it 😓)

```suggestion
      [`:matches(ExportNamedDeclaration > FunctionDeclaration, ExportNamedDeclaration > VariableDeclaration > VariableDeclarator) > Identifier.id[name="load"]`]:
```

I think the current selector has some false positives.

```js
export function foo(load) {}
export const foo = load
export function fn() {
  function load() {}
  const load = () => {}
}
```

Reply:
Ohhhhhh! fixed!

https://github.com/ota-meshi/eslint-plugin-svelte/pull/281/commits/a9030270b4cc7e9933a8f316244d841088a85e87